src/app/qgisapp.cpp

```diff
@@ -12811,7 +12821,17 @@ bool QgisApp::addRasterLayers( QStringList const &fileNameQStringList, bool guiW
       QFileInfo myFileInfo( *myIterator );
 
       // try to create the layer
-      QgsRasterLayer *layer = addRasterLayerPrivate( *myIterator, myFileInfo.completeBaseName(),
+      // set the layer name to the file base name...
+      QString layerName = myFileInfo.completeBaseName();
+
+      // ...unless the layer uri matches "GPKG:filePath:layerName" and layerName differs from the file base name
+      QStringList layerUriSegments = myIterator->split( QLatin1String( ":" ) );
+      if ( layerUriSegments.count() == 3 && layerUriSegments[ 0 ] ==  QLatin1String( "GPKG" ) && layerUriSegments[ 2 ] != layerName )
```

Comment:
either make layerUriSegments const, or use `.at( 0 )`, etc for efficiency

Reply:
Done. Not sure what is better, so I made it const.